mlflow/utils/uri.py

```diff
@@ -93,12 +98,15 @@ def add_databricks_profile_info_to_artifact_uri(artifact_uri, databricks_profile
             netloc = 'databricks'
         else:
             (profile, key_prefix) = get_db_info_from_uri(databricks_profile_uri)
-            if ':' in profile:
-                raise MlflowException("Unsupported Databricks profile name: %s." % profile +
-                                      " Profile names cannot contain ':'.")
-            if key_prefix and '/' in key_prefix:
-                raise MlflowException("Unsupported Databricks profile key prefix: %s."
-                                      % key_prefix + " Key prefixes cannot contain '/'.")
+            # Both profile and key_prefix should not contain special chars for URIs, like '/'
+            # and ':'.
+            for c in ['/', ':']:
```

Comment:
I don't think it's actually possible given the current implementation to get a `/` into either of these, since that would mean that it falls into the `path` when being parsed by the URI parser, but logically, I think it's probably still true that from a validation perspective, we want to make sure neither of those chars happens to exist.

Reply:
Why do we need to enforce that ':' is not part of the prefix? It looks like actually you cannot have either character in the secret scope name, but the key name is not as restrictive: https://docs.databricks.com/security/secrets/secret-scopes.html#overview (also we don't enforce the same restriction in other uri functions e.g. `get_databricks_profile_uri_from_artifact_uri`)

Reply:
The logic for this was that I came across scenarios like this:

Original DBFS path: 'dbfs:/path/a/b', Databricks Profile URI: 'databricks://not:legit:auth` which would result in dbfs:/not:legit:auth@databricks. 

You pointed out though, that, if we document then the scope isn't allowed to have a `:`, then the only possible way to parse this would be to include the second `:` in the second portion (the prefix)